src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.db.StorageImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy strategy;
+    private Storage storage;
+
+    public ShopServiceImpl(OperationStrategy strategy) {
+        this.strategy = strategy;
+        this.storage = new StorageImpl();
+    }
+
+    @Override
+    public Storage process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            try {
+                strategy.chooseHandler(transaction).handle(storage, transaction);
+            } catch (RuntimeException e) {
+                System.err.println("Error processing transaction: " + e.getMessage());
+            }
+        }
+        return storage;
+    }
+}
```

Comment:
It seems like the ShopService implementation is directly creating a new instance of StorageImpl. According to the checklist, services should be independent and called in the main() method. Consider injecting the Storage dependency through the constructor or a setter method to adhere to the Dependency Inversion principle.